lib/scala/kg/src/main/scala/stores/kg/KgNodeSort.scala

```diff
@@ -0,0 +1,15 @@
+package stores.kg
+
+import models.kg.KgNode
+import stores.SortDirection
+import stores.SortDirection.SortDirection
+import stores.kg.KgNodeField.KgNodeField
+
+final case class KgNodeSort(field: KgNodeField, direction: SortDirection)
+
+object KgNodeSort {
+  def sort(nodes: List[KgNode], sorts: List[KgNodeSort]): List[KgNode] = {
+    if (sorts.isEmpty) nodes
+    else KgNodeSort.sort(nodes.sortBy(node => KgNodeField.getFieldString(node, sorts.last.field))(if (sorts.last.direction == SortDirection.Ascending) Ordering[String] else Ordering[String].reverse), sorts.init)
```

Comment:
If this is the only place getFieldString is used, then make it a private method here. You need to treat the fields differently for ordering purposes, anyway. pageRank should be numerically compared, lists should be sorted and then compared, a string should be compared as-is.

KgNodeSort.sort is only used in MemKgQueryStore, anyway, so move this there.

Reply:
Gotcha that makes sense. Am experimenting with Lucene sort to replace this